toucan_connectors/google_sheets_2/google_sheets_2_connector.py

```diff
@@ -54,20 +57,41 @@ def get_form(cls, connector: 'GoogleSheets2Connector', current_config):
 Secrets = Dict[str, Any]
 
 
+class GoogleSheets2HiddenProperties(BaseModel):
```

Comment:
I don't really see the value of using a Pydantic BaseModel here. Their usage is to validate/enforce a model, when some data comes from an uncertain source, like a form filled by someone, or an unknown database without a schema.
I wonder if a very simple dataclass would be enough here.